src/main/groovy/au/org/ala/ecodata/reporting/CSProjectXlsExporter.groovy

```diff
@@ -158,22 +165,97 @@ class CSProjectXlsExporter extends ProjectExporter {
         List activityList = []
         activityList.addAll(activityIds)
 
-        List<Map> activities = []
+        int batchSize = 100
+        int processed = 0
+        def count = batchSize
+
+        int batchedActivitiesCount = 0
+
+        List<String> headers = []
+        headers.addAll(surveyHeaders)
+
+        OutputModelProcessor processor = new OutputModelProcessor()
+
+        Set<String> uniqueOutputs = [] as HashSet<String>
+
+        List<Map> batchedActivities = []
         // BioCollect currently doesn't use form versioning so just get the latest version of the form.
         ActivityForm form = activityFormService.findActivityForm(projectActivity.pActivityFormName)
         if (activityIds == null || activityIds.isEmpty()) {
-            activities = activityService.findAllForProjectActivityId(projectActivity.projectActivityId)
-        } else {
-            activities = activityService.findAllForActivityIdsInProjectActivity(activityList, projectActivity.projectActivityId)
+            def id = new MinKey()
```

Comment:
Take this comment for what it's worth as I don't fully understand what is trying to be achieved here, but won't gt "<any field>" $minKey always return true by design?   The reason I am questioning it is during other mongo upgrades the API has changed so any direct mongo dependencies result in extra work.  It's also easy to introduce bugs mixing mongo direct queries with GORM (the grails 5 upgrade ran into a bug like this).